### ISSUE TITLE
[Feature Request] select folder

### ISSUE THREAD

**Issue by happyreacer** [keeporsweep/keeporsweep#7]
It makes sence when you have a butten or so on, direct in the app to browse in your data for select a folder

**Comment by jancborchardt** [comment]
Yep, we should have a settings icon in the bottom left to reveal a settings area where the base folder can be set.

Alternatively we also heard from people who want to exclude specific folders, like important documents or so.

**Comment by aproposnix** [comment]
This would be useful for me as well.  a use case could include: 
1. User [right clicks](https://apps.nextcloud.com/apps/files_rightclick) on a folder
2. from the right click menu user selects 'clean' 
3. KS is opened in the folder (only for the contents of that folder - maybe for sub-folders as well?)

Another cool thing would be to hook into the image gallery so that from there, you can call KS directly from the gallery. Maybe a just button?

**Comment by deidyomega** [comment]
I'd like to +1 this.

I have a few programs in my nextcloud that have 100's of files.

I'd like to limit it's scope, or start it against a directory.  Namely I'd limit it to auto uploads from my phones camera.  Right now the app is useless because it never gets past the program directories.

**Comment by jancborchardt** [comment]
@deidyomega @aproposnix are you maybe interested in contributing this as a fix? :) Any help is appreciated.

**Comment by deidyomega** [comment]
I've never messed with nextcloud plugins, but I'll clone it and see if I can fix the issue.  I'll submit a PR should I solve it.

My approach will be to add a context menu to the three dots besides every folder that starts keep or sweep within that folder (and subfolders).

This will allow the 'standard' approach to keep working as it does, and allow us weird people to use specific directories.

If there's any issue with my idea please feel free to tell me why I'm an idiot!  I love advice!

**Comment by jancborchardt** [comment]
@deidyomega cool! I'd prefer if we don't inject stuff into Files as it clutters up the interface. As mentioned above the proposed design is:

> Yep, we should have a settings icon in the bottom left to reveal a settings area where the base folder can be set.

**Comment by deidyomega** [comment]
Whoops didn't see that somehow.  Okay, sounds good to me.  I'll start on it.

**Comment by deidyomega** [comment]
Hey, I'm struggling here.

The style.scss file looks like the right spot to be editing, but I don't see how/where it's compiled, and the changes aren't shown on the app.

How do I make changes to the scss and get them to show on the app?

**Comment by deidyomega** [comment]
Okay, so the javascript is good, the UI is poor at best.  Settings updates using the `OC.dialogs.prompt` which I really dislike from a UI perspective.

The only thing I couldn't get, was getting the settings on the bottom left.  Also, I was thinking for UI, maybe an accordion for settings? 

A couple of questions along that vein of thought: 
1. How can we save settings?
2. Since we are adding a settings panel, is there anything else we want in there?

I'd love feedback:
https://github.com/deidyomega/keeporsweep/tree/issue7

**Comment by jancborchardt** [comment]
I'm in the process of rewriting the interface in Vue, see https://github.com/keeporsweep/keeporsweep/tree/vue?files=1

Unfortunately it takes some time, and while I'm Nextcloud designer I don't know so much about the development unfortunately. :/

**Comment by deidyomega** [comment]
Well my day job is programming, but I write APIs and manage my company's statistic engine.  So ui development isn't my area either :(. 

Should I hold off until you complete your rewrite?  Or should I update my pr and let you finish creating the settings area once you get further in your rewrite?

**Comment by aproposnix** [comment]
> @deidyomega @aproposnix are you maybe interested in contributing this as a fix? :) Any help is appreciated.
Sorry, no coding skills here. Just ideas :)

**Comment by jancborchardt** [comment]
> Should I hold off until you complete your rewrite? Or should I update my pr and let you finish creating the settings area once you get further in your rewrite?

@deidyomega not sure to be honest, as the Vue rewrite is also a bit up in the air. We were focusing on the Twitter implementation recently https://github.com/keeporsweep/keeporsweep-twitter/ and it uses a yet a different UI.

If there is a folder picker at this stage, I would say it should be as simple as possible. Maybe just an element on the bottom left, by default saying "📁 Choose folder" and when you open it it opens a Nextcloud filepicker. Then when you picked the folder it shows only stuff from there and its subfolders and says "📁 Foldername" instead.

**Comment by Toastyyy3** [comment]
I'd also be interested in it as some folder contain important work or university related stuff and I don't want to accidentally sweep an image from there so for my specific use case a "exclude folder" function would suffice. But selecting folders is also fine of course
